mars/services/task/supervisor/tests/test_task_manager.py

```diff
@@ -686,3 +695,66 @@ async def test_dump_subtask_graph(actor_pool):
     pdf_path = os.path.join(tempfile.gettempdir(), f"mars-{task_id}.pdf")
     if os.path.exists(pdf_path):
         os.remove(pdf_path)
+
+
+@pytest.mark.asyncio
+async def test_collect_task_info(actor_pool):
+    (
+        execution_backend,
+        pool,
+        session_id,
+        meta_api,
+        lifecycle_api,
+        storage_api,
+        manager,
+    ) = actor_pool
+
+    df = md.DataFrame(mt.random.rand(20, 20, chunk_size=5))
+    graph = TileableGraph([df.data])
+    next(TileableGraphBuilder(graph).build())
+    task_id = await manager.submit_tileable_graph(
+        graph,
+        fuse_enabled=True,
+        extra_config={"collect_task_info": True},
+    )
+    await manager.wait_task(task_id)
+    yaml_root_dir = os.path.join(tempfile.tempdir, "mars_temp_yaml")
```

Comment:
Use a context manager to create temporary directory to make sure the directory would be removed on completion.